### ISSUE TITLE
Translations: injected text should be translatable

### ISSUE THREAD

**Issue by humitos** [readthedocs/addons#12]
We are generating and injecting some HTML code on the page. How are we going to make that text translatable?

* https://guillaumepotier.github.io/gettext.js/
* https://www.i18next.com/
* http://messageformat.github.io/Jed/

**Comment by agjohnson** [comment]
Making strings translated isn't a big jump, there is tooling here that we're not yet using.

The problem here is a bit deeper though, and the larger issue will be in _using_ the translations. We really should give the user a localized response from the API, based on the reader's browser language or documentation translation language, and we need this JS to come to the same conclusion about the language to use for translated strings.

**Comment by humitos** [comment]
> We really should give the user a localized response from the API, based on the reader's browser language or documentation translation language

We are not returning strings to be read by the user from the API. These strings are in Javascript code now.

**Comment by agjohnson** [comment]
Yup, understood. I'm mentioning the API response in addition to this because it's related.

This library has some native translated strings only used in this library (ie "Loading"), but we will also end up with strings translated through the backend response as we fill out the API data structure more (ie "Pull request 1234"), and there will be some strings that _could_ come from the backend if we wanted (ie "This documentation is for an outdated version ``latest``").

How JS translations are recommended in the Django world is to gettext translate the JavaScript sources, but then the Django application serves a dedicated view/URL that responds with the localized strings (JavaScriptCatalog). This guide shows everything working together:

https://localizely.com/blog/django-i18n-tutorial/

I'm not using this pattern on the dashboard yet, but it's feeling like a better pattern the more I avoid it. 

I'm not sure how applicable this particular pattern is to this library yet though. I'm not suggesting it for native strings to this library though.

**Comment by agjohnson** [comment]
It's not a target for this initial pass at this work, but the next step here would be outputting gettext catalogs for this library. We can decide _how to use the catalogs_ later, but likely will need them either way.

**Comment by agjohnson** [comment]
I was going to bring this up this week. This repo could be a good fit if we want to try using Crowdin to push translation management to GitHub. Translation volume is low and it's new.

On the tooling front, I just want to ensure that the translation tools/packages we use here are the same we use on the dashboard, so we don't have differing tech. The dashboard is not doing any translation in JS yet and pushing all of this to Django and templates instead, which is not great.

But the JS tech is largely not great here either. Most solutions just request or import localized strings from a separate JS or JSON file.

We can visit the API localization later. It's less of an issue, though there are strings in there we'll have to do something with:

- https://github.com/readthedocs/addons/blob/2dfab0c84f6cd0ec6e055466fd6ae5bd8d24d1a1/public/_/readthedocs-addons.json#L132
- https://github.com/readthedocs/addons/blob/2dfab0c84f6cd0ec6e055466fd6ae5bd8d24d1a1/public/_/readthedocs-addons.json#L149
